### PR TITLE
Milestones: Implement uml modeling

### PR DESCRIPTION
**Pull request by geekelo**
Implement UML to reflect association, aggregation, and composition

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,12 +1,14 @@
-require_relative 'person'
+class Student
```
Comment (t-yanick) [inline student.rb]: - [ ] Great job working on this milestone. Student class should inherit from Person. You seem to have done the correct thing and deleted it.
`class Student < Person`

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (t-yanick) [inline book.rb:13]: - [ ] You are missing a `method for adding rentals`. That can be achieved as in the example below:
```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Rental
+  attr_accessor :date
+  attr_reader :person, :book
```
Comment (t-yanick) [inline rental.rb]: - [ ] You should have Getters and Setters for @book and @person instance variables. So you can use `attr_accessor`

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @geekelo , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Relationships between classes correctly setup
:heavy_check_mark: Descriptive Readme file
:heavy_check_mark: Use of correct gitflow
:heavy_check_mark: Linters all passing

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by algerina — APPROVED** [review]
Hi @geekelo  ,

Great job so far!  

May I highlight the following: 
- [x]  Well done implementing the previews TSE's requirements👏  ✔️ 
- [x] Your code is clean and well organized✔️



## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:


 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

**Pull request merged**